### PR TITLE
feat(codewhisperer): neighborfiles for new context fetching strategy

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Problem
Q science team proposed a new crossfile context fetching approach which will utilize "neighborfiles" of the file where users trigger `generateCompletion`

Definition
```

/**
 *     1. A: root/util/context/a.ts
 *     2. B: root/util/b.ts
 *     3. C: root/util/service/c.ts
 *     4. D: root/d.ts
 *     5. E: root/util/context/e.ts
 *     6. F: root/util/foo/bar/baz/f.ts
 *
 *   neighborfiles(A) = [B, E]
 *   neighborfiles(B) = [A, C, D, E]
 *   neighborfiles(C) = [B,]
 *   neighborfiles(D) = [B,]
 *   neighborfiles(E) = [A, B]
 *   neighborfiles(F) = []
 *
 *  number values denote file distance between the 2 files
 *       A B C D E F
 *   A  x 1 2 2 0 4
 *   B  1 x 1 1 1 3
 *   C  2 1 x 2 2 4
 *   D  2 1 2 x 2 4
 *   E  0 1 2 2 x 4
 *   F  4 3 4 4 4 x
 */
```

## Solution

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue [hunk 1]
```diff
@@ -447,15 +447,15 @@ export default defineComponent({
...
+                .then((response) => {
                     this.initialData.checkAccessNotGrantedActionsTextArea = response.actions
                         ? response.actions.toString()
                         : ''
                     this.initialData.checkAccessNotGrantedResourcesTextArea = response.resources
                         ? response.resources.toString()
                         : ''
                 })
-                .catch(err => console.log(err))
+                .catch((err) => console.log(err))
```
Comment (Will-ShaoHua) [inline packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue:458]: changes in this file seem to be made by linter

Comment (kevluu-aws) [reply packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue:458]: ^ Changes pertaining to A2 looks fine, reviewed and deferring approval to Toolkits

**Review by andrewyuq — APPROVED** [review]

File: packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue (hunk 1 above)
Comment (nkomonen-amazon) [reply packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue:458]: This is a weird bug and we're not sure why it happens. But this is fine for now.

**Review by nkomonen-amazon — APPROVED** [review]

**Comment by justinmk3** [comment]
Is there a user-facing improvement we can mention in the changelog? E.g. "Q context now has improved context based on nearby, related files."

**Comment by Will-ShaoHua** [comment]
>  please use a valid PR title (check the failing CI job)
    Is there a user-facing improvement we can mention in the changelog? E.g. "Q context now has improved context based on nearby, related files."

ack, since it's targeting to a feature branch and I think we will do changelog once everything is ready

**Pull request merged**